### PR TITLE
[Package Add] Add webapp-manager

### PR DESCRIPTION
**Pull request by Zahrun**
Package Add

### CODE REVIEW

**Comment by Zahrun** [comment]
Sorry I don't know how to add the "Package Add" label to the PR

**Comment by wizard-28** [comment]
Just name you PR with the appropriate label, so we can identify it without labelling it.

File: packages/webapp-manager/webapp-manager.pacscript [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name="webapp-manager"
+version="1.1.2"
+url="https://github.com/linuxmint/webapp-manager/archive/refs/tags/$version.tar.gz"
+build_depends="git"
+depends="python3-bs4 python3-configobj python-gobject python-pillow python3-setproctitle tldextract dconf-cli python3-xapp gir1.2-xapp-1.0"
+breaks=""
+replace=""
```
Comment (wizard-28) [inline packages/webapp-manager/webapp-manager.pacscript:7]:
```suggestion

```
Empty fields remove them

File: packages/webapp-manager/webapp-manager.pacscript [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+description="Run websites as if they were apps"
+hash="07458737b98ac3058ad0ad72345586811b3dff868b5607f8153f8a8baa754f47"
+optdepends=()
+ppa=()
```
Comment (wizard-28) [inline packages/webapp-manager/webapp-manager.pacscript:11]:
```suggestion

```
Empty arrays remove them

File: packages/webapp-manager/webapp-manager.pacscript [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+maintainer="Zahrun <Zahrun@github.com>"
+pacdeps=()
+patch=()
```
Comment (wizard-28) [inline packages/webapp-manager/webapp-manager.pacscript:14]:
```suggestion

```
Empty arrays remove them

**Review by wizard-28 — CHANGES_REQUESTED** [review]
I get this error while installing it
```
# WARNING: the second sed below will only works correctly with the languages that don't contain "-"
for file in `ls po/*.po`; do \
	lang=`echo $file | sed 's@po/@@' | sed 's/\.po//' | sed 's/webapp-manager-//'`; \
	install -d usr/share/locale/$lang/LC_MESSAGES/; \
	msgfmt -o usr/share/locale/$lang/LC_MESSAGES/webapp-manager.mo $file; \
done \
```

And this error while running `webapp-manager`
```
Traceback (most recent call last):
  File "/usr/lib/webapp-manager/webapp-manager.py", line 20, in <module>
    from common import _async, idle, WebAppManager, Browser, download_favicon, ICONS_DIR, BROWSER_TYPE_FIREFOX
  File "/usr/src/pacstall/webapp-manager/usr/lib/webapp-manager/common.py", line 260, in <module>
    from PIL import Image
ModuleNotFoundError: No module named 'PIL'
```

Here's the full log (~ and ❯ are my shell prompts ignore them)
```
~
❯ sudo pacstall -Il webapp-manager
Do you want to view the pacscript first [y/N]
Do you want to edit the pacscript [y/N]
[+] INFO: Sourcing pacscript
[+] INFO: Running checks
[+] INFO: URL exists
[+] INFO: webapp-manager requires git to install
Do you want to remove them after installing webapp-manager [y/N]
[+] INFO: Installing dependencies
[+] INFO: Retrieving packages
1.1.2.tar.gz                          [  <=>                                                         ]  79.67K   352KB/s    in 0.2s
Building the mo files
# WARNING: the second sed below will only works correctly with the languages that don't contain "-"
for file in `ls po/*.po`; do \
	lang=`echo $file | sed 's@po/@@' | sed 's/\.po//' | sed 's/webapp-manager-//'`; \
	install -d usr/share/locale/$lang/LC_MESSAGES/; \
	msgfmt -o usr/share/locale/$lang/LC_MESSAGES/webapp-manager.mo $file; \
done \

[+] INFO: Installing
[+] INFO: Symlinking files
[+] INFO: Storing pacscript
~ took 12s
❯ webapp-manager
~
❯ Traceback (most recent call last):
  File "/usr/lib/webapp-manager/webapp-manager.py", line 20, in <module>
    from common import _async, idle, WebAppManager, Browser, download_favicon, ICONS_DIR, BROWSER_TYPE_FIREFOX
  File "/usr/src/pacstall/webapp-manager/usr/lib/webapp-manager/common.py", line 260, in <module>
    from PIL import Image
ModuleNotFoundError: No module named 'PIL'
```

File: packages/webapp-manager/webapp-manager.pacscript [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+build() {
+	make buildmo
+}
+
+install() {
+	installdir=/usr/src/pacstall/webapp-manager
+	mkdir -p $installdir
+	cp -r usr etc $installdir
```
Comment (wizard-28) [inline packages/webapp-manager/webapp-manager.pacscript:29]: What does this line do?

**Comment by Zahrun** [comment]
> Just name you PR with the appropriate label, so we can identify it without labelling it.
Well noted.

I hided the build warning and fixed the error, please check again.

I created a clean VM to test the package, I'll test the next ones in there before submitting.

File: packages/webapp-manager/webapp-manager.pacscript (hunk 4 above)
Comment (Zahrun) [reply packages/webapp-manager/webapp-manager.pacscript:25]: See https://github.com/linuxmint/webapp-manager/issues/102

**Review by wizard-28 — COMMENTED** [review]
There were no building errors this time, but while running I encountered the same error
Here is it
```
~
❯ Traceback (most recent call last):
  File "/usr/lib/webapp-manager/webapp-manager.py", line 20, in <module>
    from common import _async, idle, WebAppManager, Browser, download_favicon, ICONS_DIR, BROWSER_TYPE_FIREFOX
  File "/usr/src/pacstall/webapp-manager/usr/lib/webapp-manager/common.py", line 260, in <module>
    from PIL import Image
ModuleNotFoundError: No module named 'PIL'
```

Here's the full log
```
~ took 2s
❯ sudo pacstall -Il webapp-manager
Do you want to view the pacscript first [y/N] y\
Do you want to edit the pacscript [y/N]
[+] INFO: Sourcing pacscript
[+] INFO: URL exists
[+] INFO: webapp-manager requires gettext to install
Do you want to remove them after installing webapp-manager [y/N] y
[+] INFO: Installing dependencies
Extracting templates from packages: 100%
Preconfiguring packages ...
Selecting previously unselected package fonts-lato.
(Reading database ... 250322 files and directories currently installed.)
Preparing to unpack .../00-fonts-lato_2.0-2_all.deb ...
Unpacking fonts-lato (2.0-2) ...
Selecting previously unselected package publicsuffix.
Preparing to unpack .../01-publicsuffix_20200303.0012-1_all.deb ...
Unpacking publicsuffix (20200303.0012-1) ...
Selecting previously unselected package xapps-common.
Preparing to unpack .../02-xapps-common_1.6.10-2_all.deb ...
Unpacking xapps-common (1.6.10-2) ...
Selecting previously unselected package libxapp1:amd64.
Preparing to unpack .../03-libxapp1_1.6.10-2_amd64.deb ...
Unpacking libxapp1:amd64 (1.6.10-2) ...
Selecting previously unselected package gir1.2-xapp-1.0.
Preparing to unpack .../04-gir1.2-xapp-1.0_1.6.10-2_amd64.deb ...
Unpacking gir1.2-xapp-1.0 (1.6.10-2) ...
Selecting previously unselected package rubygems-integration.
Preparing to unpack .../05-rubygems-integration_1.16_all.deb ...
Unpacking rubygems-integration (1.16) ...
Selecting previously unselected package rake.
Preparing to unpack .../06-rake_13.0.1-4_all.deb ...
Unpacking rake (13.0.1-4) ...
Selecting previously unselected package ruby-minitest.
Preparing to unpack .../07-ruby-minitest_5.13.0-1_all.deb ...
Unpacking ruby-minitest (5.13.0-1) ...
Selecting previously unselected package ruby-net-telnet.
Preparing to unpack .../08-ruby-net-telnet_0.1.1-2_all.deb ...
Unpacking ruby-net-telnet (0.1.1-2) ...
Selecting previously unselected package ruby-power-assert.
Preparing to unpack .../09-ruby-power-assert_1.1.7-1_all.deb ...
Unpacking ruby-power-assert (1.1.7-1) ...
Selecting previously unselected package ruby-test-unit.
Preparing to unpack .../10-ruby-test-unit_3.3.5-1_all.deb ...
Unpacking ruby-test-unit (3.3.5-1) ...
Selecting previously unselected package ruby-xmlrpc.
Preparing to unpack .../11-ruby-xmlrpc_0.3.0-2_all.deb ...
Unpacking ruby-xmlrpc (0.3.0-2) ...
Selecting previously unselected package libruby2.7:amd64.
Preparing to unpack .../12-libruby2.7_2.7.0-5ubuntu1.4_amd64.deb ...
Unpacking libruby2.7:amd64 (2.7.0-5ubuntu1.4) ...
Selecting previously unselected package ruby2.7.
Preparing to unpack .../13-ruby2.7_2.7.0-5ubuntu1.4_amd64.deb ...
Unpacking ruby2.7 (2.7.0-5ubuntu1.4) ...
Selecting previously unselected package ruby.
Preparing to unpack .../14-ruby_1%3a2.7+1_amd64.deb ...
Unpacking ruby (1:2.7+1) ...
Selecting previously unselected package ruby-json.
Preparing to unpack .../15-ruby-json_2.3.0+dfsg-1build1_amd64.deb ...
Unpacking ruby-json (2.3.0+dfsg-1build1) ...
Selecting previously unselected package gist.
Preparing to unpack .../16-gist_5.0.0-4_all.deb ...
Unpacking gist (5.0.0-4) ...
Selecting previously unselected package javascript-common.
Preparing to unpack .../17-javascript-common_11_all.deb ...
Unpacking javascript-common (11) ...
Selecting previously unselected package libjs-jquery.
Preparing to unpack .../18-libjs-jquery_3.3.1~dfsg-3_all.deb ...
Unpacking libjs-jquery (3.3.1~dfsg-3) ...
Selecting previously unselected package python3-soupsieve.
Preparing to unpack .../19-python3-soupsieve_1.9.5+dfsg-1_all.deb ...
Unpacking python3-soupsieve (1.9.5+dfsg-1) ...
Selecting previously unselected package python3-bs4.
Preparing to unpack .../20-python3-bs4_4.8.2-1_all.deb ...
Unpacking python3-bs4 (4.8.2-1) ...
Selecting previously unselected package python3-configobj.
Preparing to unpack .../21-python3-configobj_5.0.6-4_all.deb ...
Unpacking python3-configobj (5.0.6-4) ...
Selecting previously unselected package python3-webencodings.
Preparing to unpack .../22-python3-webencodings_0.5.1-1ubuntu1_all.deb ...
Unpacking python3-webencodings (0.5.1-1ubuntu1) ...
Selecting previously unselected package python3-html5lib.
Preparing to unpack .../23-python3-html5lib_1.0.1-2_all.deb ...
Unpacking python3-html5lib (1.0.1-2) ...
Selecting previously unselected package python3-lxml:amd64.
Preparing to unpack .../24-python3-lxml_4.5.0-1ubuntu0.3_amd64.deb ...
Unpacking python3-lxml:amd64 (4.5.0-1ubuntu0.3) ...
Selecting previously unselected package python3-requests-file.
Preparing to unpack .../25-python3-requests-file_1.4.3-2_all.deb ...
Unpacking python3-requests-file (1.4.3-2) ...
Selecting previously unselected package python3-tldextract.
Preparing to unpack .../26-python3-tldextract_2.2.1-1_all.deb ...
Unpacking python3-tldextract (2.2.1-1) ...
Selecting previously unselected package tree.
Preparing to unpack .../27-tree_1.8.0-1_amd64.deb ...
Unpacking tree (1.8.0-1) ...
Selecting previously unselected package hddtemp.
Preparing to unpack .../28-hddtemp_0.3-beta15-53_amd64.deb ...
Unpacking hddtemp (0.3-beta15-53) ...
Selecting previously unselected package inxi.
Preparing to unpack .../29-inxi_3.0.38-1-0ubuntu1_all.deb ...
Unpacking inxi (3.0.38-1-0ubuntu1) ...
Selecting previously unselected package lm-sensors.
Preparing to unpack .../30-lm-sensors_1%3a3.6.0-2ubuntu1_amd64.deb ...
Unpacking lm-sensors (1:3.6.0-2ubuntu1) ...
Selecting previously unselected package mesa-utils.
Preparing to unpack .../31-mesa-utils_8.4.0-1build1_amd64.deb ...
Unpacking mesa-utils (8.4.0-1build1) ...
Selecting previously unselected package python3-setproctitle:amd64.
Preparing to unpack .../32-python3-setproctitle_1.1.10-1ubuntu1_amd64.deb ...
Unpacking python3-setproctitle:amd64 (1.1.10-1ubuntu1) ...
Setting up javascript-common (11) ...
Setting up fonts-lato (2.0-2) ...
Setting up ruby-power-assert (1.1.7-1) ...
Setting up xapps-common (1.6.10-2) ...
Setting up inxi (3.0.38-1-0ubuntu1) ...
Setting up rubygems-integration (1.16) ...
Setting up python3-webencodings (0.5.1-1ubuntu1) ...
Setting up ruby-minitest (5.13.0-1) ...
Setting up python3-configobj (5.0.6-4) ...
Setting up ruby-test-unit (3.3.5-1) ...
Setting up mesa-utils (8.4.0-1build1) ...
Setting up ruby-net-telnet (0.1.1-2) ...
Setting up tree (1.8.0-1) ...
Setting up python3-setproctitle:amd64 (1.1.10-1ubuntu1) ...
Setting up python3-html5lib (1.0.1-2) ...
Setting up lm-sensors (1:3.6.0-2ubuntu1) ...
Created symlink /etc/systemd/system/multi-user.target.wants/lm-sensors.service → /lib/systemd/system/lm-sensors.service.
Setting up python3-lxml:amd64 (4.5.0-1ubuntu0.3) ...
Setting up hddtemp (0.3-beta15-53) ...
Setting up python3-requests-file (1.4.3-2) ...
Setting up libjs-jquery (3.3.1~dfsg-3) ...
Setting up ruby-xmlrpc (0.3.0-2) ...
Setting up publicsuffix (20200303.0012-1) ...
Setting up python3-soupsieve (1.9.5+dfsg-1) ...
Setting up python3-tldextract (2.2.1-1) ...
/usr/lib/python3/dist-packages/tldextract/cli.py:51: SyntaxWarning: "is" with a literal. Did you mean "=="?
  elif len(args.input) is 0:
Setting up python3-bs4 (4.8.2-1) ...
Setting up libruby2.7:amd64 (2.7.0-5ubuntu1.4) ...
Setting up ruby2.7 (2.7.0-5ubuntu1.4) ...
Setting up ruby (1:2.7+1) ...
Setting up ruby-json (2.3.0+dfsg-1build1) ...
Setting up rake (13.0.1-4) ...
Setting up gist (5.0.0-4) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for libglib2.0-0:amd64 (2.64.2-1~fakesync1) ...
Processing triggers for libc-bin (2.31-0ubuntu9) ...
Processing triggers for systemd (245.4-4ubuntu3.1pop0~1590695674~20.04~eaac747) ...
Processing triggers for man-db (2.9.1-1) ...
Setting up libxapp1:amd64 (1.6.10-2) ...
Processing triggers for fontconfig (2.13.1-2ubuntu3) ...
Setting up gir1.2-xapp-1.0 (1.6.10-2) ...
Processing triggers for libc-bin (2.31-0ubuntu9) ...
[+] INFO: Retrieving packages
1.1.2.tar.gz                          [ <=>                                                          ]  79.67K  --.-KB/s    in 0.1s
[+] INFO: Installing
[+] INFO: Symlinking files
[+] INFO: Storing pacscript
~ took 32s
❯ webapp-manager
~
❯ Traceback (most recent call last):
  File "/usr/lib/webapp-manager/webapp-manager.py", line 20, in <module>
    from common import _async, idle, WebAppManager, Browser, download_favicon, ICONS_DIR, BROWSER_TYPE_FIREFOX
  File "/usr/src/pacstall/webapp-manager/usr/lib/webapp-manager/common.py", line 260, in <module>
    from PIL import Image
ModuleNotFoundError: No module named 'PIL'
```

File: packages/webapp-manager/webapp-manager.pacscript (hunk 4 above)
Comment (wizard-28) [reply packages/webapp-manager/webapp-manager.pacscript:25]: Okay

**Comment by Zahrun** [comment]
Ok, I fixed the dependency. This package comes preinstalled in Ubuntu. Which version do you use to do the tests ?

**Comment by wizard-28** [comment]
> Ok, I fixed the dependency. This package comes preinstalled in Ubuntu. Which version do you use to do the tests ?

I use Pop!_OS 20.04

**Review by wizard-28 — COMMENTED** [review]
Okay it works now.

 Finally you need to edit the [`packagelist`](https://github.com/Zahrun/pacstall-programs/blob/master/packagelist) file, and put this app's name alphabetically in it.

And since you are a first-time contributer, you are requested to join the Pacstall Organization on GitHub, so we can @mention you, add you as a reviewer and other things.

**Comment by Zahrun** [comment]
> Okay it works now.
> 
> Finally you need to edit the [`packagelist`](https://github.com/Zahrun/pacstall-programs/blob/master/packagelist) file, and put this app's name alphabetically in it.
> 
> And since you are a first-time contributer, you are requested to join the Pacstall Organization on GitHub, so we can @mention you, add you as a reviewer and other things.

Good, thank you for the great project.

**Review by wizard-28 — APPROVED** [review]
Thanks, merging this.

**Pull request merged**